include/rapidjson/reader.h

```diff
@@ -1574,8 +1576,12 @@ class GenericReader {
         if (Consume(s, '.')) {
             decimalPosition = s.Length();
 
-            if (RAPIDJSON_UNLIKELY(!(s.Peek() >= '0' && s.Peek() <= '9')))
-                RAPIDJSON_PARSE_ERROR(kParseErrorNumberMissFraction, s.Tell());
+            if (RAPIDJSON_UNLIKELY(!(s.Peek() >= '0' && s.Peek() <= '9'))) {
+                if (!(parseFlags & kParsePartialFractionFlag))
+                    RAPIDJSON_PARSE_ERROR(kParseErrorNumberMissFraction, s.Tell());
+                else if (!(s.Peek() == ',' || s.Peek() == '}' || s.Peek() == ']' || s.Peek() == '\0' || s.Peek() == 'e' || s.Peek() == 'E'))
```

Comment:
I think this should not be needed here. This function should parse as many characters for a JSON number type as it could. It should not involve other characters such as `,}]\0`.

Reply:
Thanks for reviewing the code.

I think it is needed to accept a missing fraction.

In the standard case, we should outright reject the number if it does not have numbers following the `.`.

In the partial fraction case, we can accept the number to finish here (i.e. check for `,}]\0`) or to be followed by an exponent but it should not accept anything else.

Without this check, the parser works correctly witht incomplete fractions but the parser fails with `kParseErrorDocumentRootNotSingular` when parsing `.a123` rather than `kParseErrorValueInvalid` which might be a bit confusing. If you think this is ok, I'll hapilly remove that check.

Reply:
I think implementing this relaxed syntax just involve change of the underlying state machine.
For `123.a`, it should take `123.` and returns a double valued `123.0`.
For `.123a`, it should take `.123` and returns a double valued `.123`.
For `.a` it should raise a parse error because there must be at least a digit when it is prefixed by `.`.

Reply:
Ok, I've removed the extra checks then.